ports/libpq/vcpkg-cmake-wrapper.cmake

```
@@ -15,27 +15,34 @@ if(PostgreSQL_FOUND AND @USE_DL@)
         endif()
     endif()
 endif()
-if(PostgreSQL_FOUND AND TARGET PostgreSQL::PostgreSQL AND "@VCPKG_LIBRARY_LINKAGE@" STREQUAL "static")
-    foreach(LIB_ITEM pgport pgcommon)
+if(PostgreSQL_FOUND AND "@VCPKG_LIBRARY_LINKAGE@" STREQUAL "static")
+    include(SelectLibraryConfigurations)
+    foreach(LIB_ITEM libpgport libpgcommon pgport pgcommon)
         find_library(PostgreSQL_${LIB_ITEM}_LIBRARY_RELEASE
             NAMES ${LIB_ITEM}
             PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib" NO_DEFAULT_PATH
         )
-        if(PostgreSQL_${LIB_ITEM}_LIBRARY_RELEASE)
+        if(TARGET PostgreSQL::PostgreSQL AND PostgreSQL_${LIB_ITEM}_LIBRARY_RELEASE)
             set_property(
                 TARGET PostgreSQL::PostgreSQL
                 APPEND PROPERTY INTERFACE_LINK_LIBRARIES "\$<\$<NOT:\$<CONFIG:DEBUG>>:${PostgreSQL_${LIB_ITEM}_LIBRARY_RELEASE}>"
             )
         endif()
+        
         find_library(PostgreSQL_${LIB_ITEM}_LIBRARY_DEBUG
             NAMES ${LIB_ITEM}
             PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug/lib" NO_DEFAULT_PATH
         )
-        if(PostgreSQL_${LIB_ITEM}_LIBRARY_DEBUG)
+        if(TARGET PostgreSQL::PostgreSQL AND PostgreSQL_${LIB_ITEM}_LIBRARY_DEBUG)
             set_property(
                 TARGET PostgreSQL::PostgreSQL
                 APPEND PROPERTY INTERFACE_LINK_LIBRARIES "\$<\$<CONFIG:DEBUG>:${PostgreSQL_${LIB_ITEM}_LIBRARY_DEBUG}>"
             )
         endif()
+        
+        if (PostgreSQL_${LIB_ITEM}_LIBRARY_RELEASE AND PostgreSQL_${LIB_ITEM}_LIBRARY_DEBUG)

```

```suggestion
        if (PostgreSQL_${LIB_ITEM}_LIBRARY_RELEASE OR PostgreSQL_${LIB_ITEM}_LIBRARY_DEBUG)
```
otherwise it won't get correct in single configuration builds.

---

@Neumann-A  Will this situation affect the result of  `select_library_configurations`?

---

Yes if only one variable is defined it will set the result to only the one library which is needed in the single configuration build case. 
This behavior is also defined in https://cmake.org/cmake/help/latest/module/SelectLibraryConfigurations.html